### PR TITLE
[Wallet SDK] Port useful changes from MGI guide to the generic Sep-24 guide

### PR DESCRIPTION
**Pull request by CassioMG**
This PR ports some useful changes from [Integrate with MoneyGram Access](https://developers.stellar.org/docs/build/apps/moneygram-access-integration-guide) specific guide to the generic [Hosted Deposit and Withdrawal](https://developers.stellar.org/docs/build/apps/wallet/sep24) guide.

This PR also fixes some small `typos` and consolidate the variable name `token` to be `authToken` (since a few places were using `token` and other places were using `authToken` for the same thing).

Preview is available here (most of the changes are on the `Typescript` tab): https://developers-pr930.previews.kube001.services.stellar-ops.com/docs/build/apps/wallet/sep24

### CODE REVIEW

File: docs/build/apps/wallet/sep24.mdx [hunk 1]
```diff
@@ -542,6 +701,12 @@ final paymentResult = await sdk.submitTransaction(transaction);
 
 Where `keypair` is the SEP-10 authenticated account. If you want to transfer funds from a different address, refer to [Changing Stellar Transfer Account](#changing-stellar-transfer-account) section.
 
+Code for submitting transactions to Stellar should be developed thoughtfully. The SDF has a documentation page dedicated to [submitting transactions and handling errors gracefully]. Here are a few things you need to keep in mind:
```
Comment (piyalbasu) [inline docs/build/apps/wallet/sep24.mdx:704]: missing a link?

**Review by piyalbasu — APPROVED** [review]

File: docs/build/apps/wallet/sep24.mdx [hunk 2]
```diff
@@ -347,18 +347,61 @@ While `Watcher` class offers powerful tracking capabilities, sometimes it's requ
 <CodeExample>
 
 ```kt
-val transaction = sep24.getTransactionBy(token, id = "transaction id")
+// "id" is the actual Anchor transaction id, all transactions should have it.
+val transaction = sep24.getTransactionBy(authToken, id = "transaction id")
+
+// "stellarTransactionId" (aka "stellar_transaction_id" on the SEP spec)
+// is the hash of the Stellar network transaction payment related to this
```
Comment (aristidesstaffieri) [inline docs/build/apps/wallet/sep24.mdx:354]: Would it make sense to expand on "is the hash of the Stellar network transaction payment"?
Do we think the reader assumes sha256 already or should we add a snippet?

**Review by aristidesstaffieri — APPROVED** [review]

File: docs/build/apps/wallet/sep24.mdx (hunk 1 above)
Comment (CassioMG) [reply docs/build/apps/wallet/sep24.mdx:704]: @piyalbasu the link is here at the bottom: https://github.com/stellar/stellar-docs/pull/930/files#diff-39e9cfa3d8b8714200fde6146f37f172556d4f1b750a333621353ac9dbd122feR788

File: docs/build/apps/wallet/sep24.mdx (hunk 2 above)
Comment (CassioMG) [reply docs/build/apps/wallet/sep24.mdx:354]: @aristidesstaffieri good point, I've added a change here: https://github.com/stellar/stellar-docs/pull/930/commits/fd92a204d27510ff467f29580c5aed8270c22429
let me know what you think - thanks!

Comment (aristidesstaffieri) [reply docs/build/apps/wallet/sep24.mdx:354]: lgtm 👍

**Pull request merged**